src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -349,6 +373,38 @@ HashSet<ITypeSymbol> typeStack
 			return MutabilityInspectionResult.NotMutable( unauditedReasonsBuilder.ToImmutable() );
 		}
 
+		private MutabilityInspectionResult InspectInterface(
+			ITypeSymbol symbol,
+			HashSet<ITypeSymbol> typeStack
+		) {
+			var typeSymbol = symbol as INamedTypeSymbol;
+
+			// Detects if the interface itself is marked with Immutable
+			var scope = typeSymbol.GetImmutabilityScope();
+
+			// There is a 1:1 correlation between TypeParameters and TypeArguments.
+			// TypeParameters is the "S", or "T" definition.
+			// TypeArguments are the actual *types* passed to S or T.
+			for (int ordinal = 0; ordinal < typeSymbol.TypeParameters.Length; ordinal++) {
+				var parameterType = typeSymbol.TypeArguments[ordinal];
+				var isToBeImmutable = IsTypeArgumentImmutable( typeSymbol.TypeParameters[ordinal], ordinal, typeSymbol );
+				var result = InspectType( parameterType );
+				if( result.IsMutable && isToBeImmutable ) {

```

Comment:
micro optimization if you break up this line, but it might make your branches more clear and easier to verify you've tested

```
if( !result.IsMutable ) {
   continue;
}

if( isToBeImmutable ) {
   return MutabilityInspectionResult.MutableType( parameterType, MutabilityCause.IsMutableTypeParameter );
}
```